docs/security.md

```diff
@@ -0,0 +1,182 @@
+
+```Dockerfile
+FROM alpine:3.21.3@sha256:a8560b36e8b8210634f77d9f7f9efd7ffa463e380b75e2e74aff4511df3ef88c
+```
+
+---
+
+## 3. General Best Practices
+
+### 3.1. Monitoring
+
+- **Audit logs** - Regularly audit the GitHub Actions logs for any signs of unusual behaviour or errors, especially in
+  workflows that involve sensitive data or production deployments.
+
+### 3.2. Minimise workflow triggers
+
+- **Control workflow triggers** - Be specific with workflow triggers. For example, avoid unnecessary workflow executions
+  for every push to any branch. Limit triggers to the events that are necessary for the workflow.
+
+```yaml
+on:
+  push:
+    branches: ["main"]
+```
+
+- **Restrict sensitive workflow runs** - Restrict sensitive workflows, especially deployment workflows, to only trusted
+  branches with strict access control to prevent malicious triggers from unauthorised users. Additionally, where
+  possible, use GitHub environments with rulesets configured to apply additional security rules for the deployment.
+
+### 3.3. Dependency management
+
+- **Automate dependency scanning** - When appropriate, use tools such as Dependabot or Renovate bot to automatically
+  monitor dependencies. This allows teams to be notified when a dependency update is available, so that the update can
+  be reviewed and then used.
+
+### 3.4 Script safety
+
+- **Avoid untrusted input in scripts** - Never interpolate untrusted input directly into shell commands. Always validate
+  and sanitise inputs from users, pull requests, or other sources before use.
+
+- **Beware of `${{ }}` expression injection** - Inputs from `${{ github.event.* }}` and other contexts may be
+  attacker-controlled (e.g. pull request titles, branch names, commit messages, inputs, etc.). Do not interpolate these
+  directly into shell scripts without validation.
+
+```yaml
+# ❌ Vulnerable to injection
+run: |
+  echo "PR title: ${{ github.event.pull_request.title }}"
+
+# ✅ Safer
+env:
+  PR_TITLE: "${{ github.event.pull_request.title }}"
+run: |
+  echo "PR title: $PR_TITLE"
+```
+
+- **Avoid `eval`, backticks, or unescaped interpolation** - Avoid using `eval`, backticks, or dynamically constructed
+  commands unless absolutely necessary. These patterns are highly prone to injection vulnerabilities.
+
+- **Avoid `curl | sh`-like patterns** - Instead, download the scripts and preferably validate the checksum prior to
+  execution.
+
+```yaml
+# ❌ Dangerous
+run: curl -sSL https://example.com/install.sh | sh
+
+# ✅ Safer
+run: |
+  curl -sSL https://example.com/install.sh -o install.sh
+  echo "$INSTALLSH_CHECKSUM" | sha256sum --check -
+  chmod +x install.sh
+  ./install.sh
+```
+
+---
+
+## 4. Reporting issues
+
+If you notice a potential security issue in an action or workflow, raise it immediately with your team lead, via
+internal security channels or by emailing [`security@hemi.xyz`](mailto:security@hemi.xyz). Quick detection and response
+help us protect from potential attacks.
```

Comment:
Should we link to https://github.com/hemilabs/.github/blob/main/SECURITY.md#reporting-a-vulnerability ?

Reply:
Yes, thanks.